boltdesignsystem/bolt
# Feature/update component generator

## Jira

http://vjira2:8080/browse/BDS-2097

## Summary

Update component generator to use latest conventions, some other minor improvements.

## Details

- Convert schema from YML to JS
  - Remove description from schema, not required as docs reference `package.json`
- Update component JS to use latest conventions, including auto-props
- Remove Polyfill loader, not required since dropping IE11 support
- Add conditional prop example to component Twig template
- Add title comment to SCSS, fix formatting
- Remove boilerplate component description. We tend to include this on every component and it doesn't add much. Make authors describe the component.
- Update Jest tests to use latest conventions
  - Update generator test snapshot
  - Remove dupe Jest test file

## How to test

- Review changes
- Tests are passing
- Checkout locally and make a component (`yarn cc`)
  - Verify tests pass on that component
